File: datasets/lua/HumanEval_27_flip_case.lua
-- For a given string, flip lowercase characters to uppercase and uppercase to lowercase.
-- >>> flip_case('Hello')
-- 'hELLO'
local function flip_case(string)
  local new_string = ''
  for i = 1, #string do
    local char = string:sub(i, i)
    if char:match('%l') then
      new_string = new_string .. char:upper()
    elseif char:match('%u') then
      new_string = new_string .. char:lower()
    else
      new_string = new_string .. char
    end
  end
  return new_string
end
-- Unit tests below

lu = require('luaunit')

function test_humaneval()
local candidate = flip_case
    lu.assertEquals(candidate(''), '')
    lu.assertEquals(candidate('Hello!'), 'hELLO!')
    lu.assertEquals(candidate('These violent delights have violent ends'), 'tHESE VIOLENT DELIGHTS HAVE VIOLENT ENDS')
end

os.exit(lu.LuaUnit.run())

File: datasets/rb/HumanEval_27_flip_case.rb
# For a given string, flip lowercase characters to uppercase and uppercase to lowercase.
# >>> flip_case('Hello')
# 'hELLO'
def flip_case(string)
  string.swapcase
end

require 'test/unit'
class TestHumanEval < Test::Unit::TestCase
  def test_flip_case
    candidate = method(:flip_case)
    assert_equal('', candidate.call(''))
    assert_equal('hELLO!', candidate.call('Hello!'))
    assert_equal('tHESE VIOLENT DELIGHTS HAVE VIOLENT ENDS', candidate.call('These violent delights have violent ends'))
  end
end
